### ISSUE TITLE
What is the future of this project?

### ISSUE THREAD

**Issue by rotexdegba** [auraphp/Aura.Sql#189]
What is the future of this project? In my opinion, it is feature complete and only needs to have the latest versions of PHP (7.3, 7.4 & 8.0 as of this writing) tested via travis or Github Actions to ensure it continues to work with the currently supported version of PHP. Will anyone be updating the CI tests for this project?

**Comment by koriym** [comment]
This repo has a CI test with php8.
https://github.com/auraphp/Aura.Sql/blob/3.x/.travis.yml#L22

**Comment by rotexdegba** [comment]
> This repo has a CI test with php8.
> https://github.com/auraphp/Aura.Sql/blob/3.x/.travis.yml#L22

There should be a php 8.0 entry below here   https://github.com/auraphp/Aura.Sql/blob/c12e9f028ab8c83e9d1daee593f5d978882c488f/.travis.yml#L9

**Comment by koriym** [comment]
Not officially supported with PHP8 at the moment.
Feel free to fix it and make a pull request. Contribution appreciated.

**Comment by brandonsavage** [comment]
I confirmed today that this package at HEAD does run on PHP 8.0.9 (all tests green). We need to do a few things to make this package great again:

* Release a 3.0.1 or 3.1.0 with PHP 8 compatibility.
* Set up GitHub actions to build the project
* Have a discussion about dropping support for unsupported or old versions of PHP (I'm thinking 7.1 and below).
* Determine what, if any, new features should be built into the package.
* I also recommend a Docker development environment that can build and run all versions of PHP we support so we can run the tests in all of them.

I'm happy to help work on this, though I'm not interested in owning it.

**Comment by kenjis** [comment]
> Have a discussion about dropping support for unsupported or old versions of PHP (I'm thinking 7.1 and below).

- The Aura 3.x series was for PHP 5.6 and later.
- The Aura 4.x series is for PHP 7.2 and later.
- I imagine a 5.x series, if you want to start one, should be for ~~8.0 (or perhaps~~ 8.1 ~~?)~~ and later.

See https://github.com/auraphp/Aura.Cli_Kernel/pull/12#issuecomment-873248524

**Comment by brandonsavage** [comment]
I think this makes a lot of sense. We'll also need to discuss end of life for ancient branches. I would suggest that once the oldest version is out of support for 3 years we consider a 2 year end of life period. That would make the Aura 2.x packages end of life in 2019, which means we would issue one final release of this package under 2.x and call it a day. 3.x would get support until December 31, 2023. What do you think?

**Comment by iansltx** [comment]
So at this point we have versions that works with 7.2-8.1 (4.x) and 8.1+ (5.x). Question is, what's preventing us from tagging those? I think 4.x can be tagged as 4.0.0 now as it's a bridge version so folks can, without deprecation warnings, run the same lib version in 8.0 and 8.1, and doesn't need any additional functionality. Just bumped our codebase to 5.x and I'd like to drop `@dev` from that if we can.

**Comment by brandonsavage** [comment]
@iansltx 4.0.0 and 5.0.0 have both been tagged and pushed. Thanks for the suggestion. Please let us know if there's anything else you need.

**Comment by tbreuss** [comment]
Is the documentation regarding all this tag and php version stuff up-to-date?

**Comment by brandonsavage** [comment]
Since these changes were internal modifications only and test updates, there was no need to change the documentation. As far as I'm aware the behavior remains the same.

**Comment by r4ndsen** [comment]
> Please let us know if there's anything else you need

yes, please adjust the return types to reflect the actual behavior of the underlying pdo object:
https://github.com/auraphp/Aura.Sql/issues/194#issuecomment-1005945179

**Comment by harikt** [comment]
Hi,

I have created a separate issue to discuss about the future of aura packages. If you guys can contribute it will be great. https://github.com/auraphp/bin/issues/13

Regarding the documentation, we need to update the bookdown links in the auraphp.github.com repo. I can do that when I get sometime.

**Comment by harikt** [comment]
@r4ndsen #194 is merged and 5.0.1 is released. This note is for reference only.

**Comment by harikt** [comment]
Hi guys, 

I guess this has been addressed to some extend. 

We will be having 3.x which will work upto 8.1 once #209 is merged. And also 5.x for those who are in 8.1 .

The tests are all running on github CI for many months now.

As this is an opensource project anyone can step in and send PR . I hope any of the members of our team will try to address it.

Thank you.

**Comment by rotexdegba** [comment]
Thank you. I will look into future contributions to this and other aura packages I am familiar with.
